src/main/kotlin/org/rust/cargo/project/model/impl/CargoProjectImpl.kt

```diff
@@ -339,7 +339,9 @@ data class CargoProjectImpl(
     }
 
     override val presentableName: String
-        get() = workingDirectory.fileName.toString()
+        get() = projectService.allPackages.singleOrNull {
```

Comment:
Let's use `lazy` here to avoid iteration over all packages on each call.
It makes cargo project sorting much slower (`O(n^2logn)` instead of `O(nlogn)`). Of course, it will be noticeable only with a lot of package but anyway